extension/opampextension/config.go

```diff
@@ -89,11 +87,38 @@ func (c *commonFields) Validate() error {
 	return nil
 }
 
+type httpFields struct {
+	commonFields `mapstructure:",squash"`
+
+	PollingInterval time.Duration `mapstructure:"polling_interval"`
+}
+
+func (h *httpFields) Validate() error {
+	if err := h.commonFields.Validate(); err != nil {
+		return err
+	}
+
+	if h.PollingInterval < 0 {
+		return errors.New("polling interval must be 0 or greater")
+	}
+
+	return nil
+}
+
+// OpAMPServer contains the OpAMP transport configuration.
+type OpAMPServer struct {
+	WS   *commonFields `mapstructure:"ws,omitempty"`
```

Comment:
```suggestion
	WS   *wsFields `mapstructure:"ws,omitempty"`
```

Reply:
Using `commonFields` here is intentional, in order for us to be able to share the same struct between WS and HTTP - the only addition for HTTP is the polling interval.

Reply:
got it